lib/WUI/link_content/prusa_link_api.cpp

```diff
@@ -64,6 +64,19 @@ optional<ConnectionState> PrusaLinkApi::accept(const RequestParser &parser) cons
         return get_only(StatusPage(Status::NoContent, parser.status_page_handling(), parser.accepts_json));
     } else if (suffix == "settings") {
         return get_only(SendStaticMemory("{\"printer\": {}}", http::ContentType::ApplicationJson, parser.can_keep_alive()));
+    } else if (remove_prefix(suffix, "v1/files").has_value()){
+        static const auto prefix = "/api/v1/files/usb";
+        static const size_t prefix_len = strlen(prefix);
+        char fname[FILE_PATH_BUFFER_LEN + prefix_len];
+        if (!parser.uri_filename(fname, sizeof fname))
+            return StatusPage(Status::NotFound, parser.status_page_handling(), parser.accepts_json);
+
+        static const auto prefix_storage = "/api/v1/files";
+        static const size_t prefix_storage_len = strlen(prefix_storage);
+        char* storage_path = &fname[0] + prefix_storage_len;
+        auto upload = GcodeUpload::start(parser, wui_uploaded_gcode, parser.accepts_json, storage_path, parser.print_after_upload);
+        return std::visit([](auto upload) -> ConnectionState { return std::move(upload); }, std::move(upload));
```

Comment:
I've noticed few things around here, mostly coming from assumption that the input would be correct / benign. This comes over the network, so we can't assume any such thing 😇

* I suspect this'll allow uploading to other places on the printer than just `/usb`.
* The `&fname[0] + ...` may point outside of the file name (not out of the buffer). There isn't much guarantee of what's behind the `\0` in there, so anyone calling a `strlen` or anything might _then_ run out of the buffer.
* That construct is probably more complex than necessary, that's the same as `fname + prefix_storage_len`.
* Conceptually, this is the right place to check for the `PUT` method. The selectors decide _what_ gets done, the `GcodeUpload` is responsible for doing it, not for choosing it. Once we want to support the `HEAD`, it'll be decided here too.